### PR TITLE
Feature/get generated reports geo 66

### PR DESCRIPTION
**Pull request by jer3k**
https://finrms.atlassian.net/jira/software/projects/GEO/boards/70/backlog?selectedIssue=GEO-66

---
Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-154-frontend.apps.silver.devops.gov.bc.ca)

### CODE REVIEW

File: frontend/src/common/apiService.js [hunk 1]
```diff
@@ -112,17 +116,16 @@ export default {
...
-function getCodes(url) {
-  return async function getCodesHandler(query) {
+  async getPublishedReports() {
     try {
-      return await apiAxios.get(url, query);
+      const resp = await apiAxios.get(ApiRoutes.PUBLISHED_REPORTS);
+      if (resp?.data) {
+        return resp.data;
+      }
+      throw new Error('Unable to fetch NAICS codes from API');
```
Comment (banders) [inline frontend/src/common/apiService.js:125]: change "NAICS codes" to "published reports"

File: frontend/src/common/apiService.js [hunk 2]
```diff
@@ -112,17 +116,16 @@ export default {
...
     } catch (e) {
-      console.log(`Failed to get from Nodejs API - ${e}`);
+      console.log(`Failed to get NAICS from API - ${e}`);
```
Comment (banders) [inline frontend/src/common/apiService.js:127]: change "NAICS" to "published reports"

File: backend/src/v1/routes/report-routes.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const reportRouter = express.Router();
+
+/**
+ * /v1/report/published
+ *
+ * Get all published reports for the company associated with the logged in user.
+ */
+reportRouter.get(
+  '/published',
```
Comment (sukanya-rath) [inline backend/src/v1/routes/report-routes.ts:16]: change the url - remove /published , add a` query parameter` called `report_status` (which can be take values either `published` or `draft`)

**Review by sukanya-rath — CHANGES_REQUESTED** [review]

**Review by banders — COMMENTED** [review]
I agree with Sukanya's suggestion, and I have no additional suggestions.  Nice job with the unit tests.

File: backend/src/v1/routes/report-routes.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const reportRouter = express.Router();
+
+/**
+ * /api/v1/report/?status=<string>
+ *     Get all published reports for the company associated with the logged in user.
+ */
+reportRouter.get(
+  '/',
+  passport.authenticate('jwt', { session: false }, undefined),
+  auth.isValidBackendToken(),
```
Comment (banders) [inline backend/src/v1/routes/report-routes.ts:17]: Sonar identified a possible bug here.  auth.isValidBackendToken() needs to be wrapped in a call to utils.asyncHandler(...).  See example in code-routes.ts.

File: backend/src/v1/services/report-service.ts [hunk 5]
```diff
@@ -664,20 +698,51 @@ const reportService = {
...
   },
+
+  /**  Return all published reports created by a company */
+  async getReportsByStatus(
+    bceidBusinessGuid: string,
+    status: enumReportStatus,
+  ) {
+    const reports = await prisma.pay_transparency_company.findFirst({
+      select: {
+        pay_transparency_report: {
```
Comment (banders) [inline backend/src/v1/services/report-service.ts:719]: Can we sort the reports by report start date (newest to oldest), then secondarily by update date (newest to oldest)?

File: frontend/src/common/apiService.js [hunk 6]
```diff
@@ -131,14 +131,23 @@ export default {
...
+      const resp = await apiAxios.get(ApiRoutes.REPORTS, {
+        params: { status: status },
+      });
+      if (resp?.data) {
+        return resp.data;
+      }
+      throw new Error('Unable to fetch published reports from API');
     } catch (e) {
-      console.log(`Failed to get from Nodejs API - ${e}`);
+      console.log(`Failed to get published reports from API - ${e}`);
```
Comment (banders) [inline frontend/src/common/apiService.js:149]: remove the word 'published' from the log statement.

**Review by banders — APPROVED** [review]

**Pull request merged**